homeassistant/helpers/entity_values.py

```diff
@@ -0,0 +1,42 @@
+"""A class to hold entity values."""
+import fnmatch
+import re
+
+from homeassistant.core import split_entity_id
+
+
+class EntityValues(object):
+    """Class to store entity id based values."""
+
+    def __init__(self, exact=None, domain=None, glob=None):
+        """Initialize an EntityConfigDict."""
+        self._cache = {}
+        self._exact = exact
+        self._domain = domain
+
+        if glob is not None:
+            glob = {re.compile(fnmatch.translate(key)): value
+                    for key, value in glob.items()}
+
+        self._glob = glob
+
+    def get(self, entity_id):
```

Comment:
The custom UI went out in 0.38, but it doesn't matter - I can change it to flat structure.

The only "core" attribute customize overrides is `hidden` that affects recorder and other places.
`friendly_name` affects UI.
Anything else affects devices.

That said, specifically for custom UI I think a better place would be a special custom_ui component that will allow to pass config to the UI. This is how I originally implemented it, but @balloob didn't like it so I felt back to using customize.

Regardless:
If there is a firm decision that object values are bad - they should be banned by schema and not surprisingly break one day.